### PR TITLE
Clean up of the light and phytoplankton growth calculations

### PR DESCRIPTION
**Pull request by charliestock**
This pull request is primarily a code clean up and improved commenting for the light and phytoplankton growth sections of the COBALT code.  It was done in part to prepare for the discussion of these sections at our next COBALT "doc and dev" meeting.  

I added some fairly extensive commenting that I am hoping could help spur the documentation part of this effort.  I also eliminated substantial parts of the code that reflected different settings tested in the COBALTv3 photoacclimation analysis which were not ultimately used.  While these eliminations were substantial, I regret to say that my comments were even more copious, resulting in a modest increase in the number of lines in the code.  We'll need to decide on the proper penalty for this!

During the cleanup process I did discover that one parameter - ml_aclm_efold - was not quite defined as intended.  It was intended to be the number of irradiance e-foldings that defined a deep mixed layer for photoacclimation.   However, it was not coded in this way.  I fixed this on line 8374 and changed the parameter value from 4.6 to 2.5 to maintain the equivalent behavior with the proper interpretation of the parameter.

Finally, since the overwhelming majority of the changes were deletions of commented out code or the addition of new comments, my testing was minimal.  Perhaps we could develop a wiki entry with a few recommended configurations for testing before submitting a pull request?  Apologies in advance for any hassles that arise, but if we could advance this pull request before the next "doc and dev", I think it will help the discussion considerably.

### CODE REVIEW

**Comment by yichengt900** [comment]
This PR addresses all the comments from reviewers and is up-to-date with the main branch. Please review it again, and feel free to provide any further comments, thanks.

**Comment by yichengt900** [comment]
Thanks, @charliestock . I can raise this PR for tomorrow's discussion. Currently, it failed the CI test. If you agree, I can work on your behalf and fix the bug in your feature branch so that I can approve and merge this PR.

**Comment by charliestock** [comment]
Hello Yi-Cheng,

I'm stuck in a meeting until 1 PM today, but lets discuss when we talk this afternoon.  I suspect this issue is something small that I can fix quickly this afternoon or evening.  Did the CI test complain about anything in particular?

**Comment by yichengt900** [comment]
@charliestock, no worries. I also suspect the error should be small. I will check it, and perhaps I will be able to fix it from my side. However, I will let you review it before we move forward."

File: generic_tracers/generic_COBALT.F90 [hunk 1]
```diff
@@ -8326,35 +8324,57 @@ subroutine generic_COBALT_update_from_source(tracer_list,Temp,Salt,rho_dzt,dzt,h
...
 
-
-    allocate(tmp_irr_band(nbands))
-    allocate(sfc_irrad(isc:iec,jsc:jec))
-    allocate(kblt(isc:iec,jsc:jec))
-    frac_sfc_irrad_aclm = 1.0/(cobalt%ml_aclm_efold*2.71828)
+    !
+    ! Calculate the underwater light field for the BGC calculations.  By default, COBALT
+    ! receives visible and near-infrared inputs.  Near infrared is currently considered
+    ! anyting with wavelengths longer than ~710 nanometers.  Note that both shortwave and
```
Comment (yichengt900) [inline generic_tracers/generic_COBALT.F90]: `anything`

File: generic_tracers/generic_COBALT.F90 [hunk 2]
```diff
@@ -8326,35 +8324,57 @@ subroutine generic_COBALT_update_from_source(tracer_list,Temp,Salt,rho_dzt,dzt,h
...
+    ! near infrared fall into the "shortwave" part of the radiation spectrum.  Following
+    ! Morel and Antoine (1994) and Sweeney et al. (2005), about 57% of the incoming irradiance
+    ! is assumed to lie within the visible range in the standard MOM6 radiation scheme
+    ! with interactive chlorophyll.    
+    ! 
+    ! In COBALTv1/v2, all of the visible wavelengths were included as photosynthically
+    ! active radiation (PAR).  This approach, however, included wavelengths shorter than  
+    ! than the ~350-400 nanometer lower bound applied for PAR.  The "par_adj" parameter
```
Comment (yichengt900) [inline generic_tracers/generic_COBALT.F90]: delete additional `than`

File: generic_tracers/generic_COBALT.F90 [hunk 3]
```diff
@@ -8427,30 +8464,30 @@ subroutine generic_COBALT_update_from_source(tracer_list,Temp,Salt,rho_dzt,dzt,h
...
 
        enddo !} k-loop
 
-       ! Calculate the instantaneous acclimation irradiance, which will be integrated over the photoacclimation
-       ! timescale (see below)
+       ! Phytoplankton in the surface mixed layer (1:kblt) aclimate to the mean daytime
+       ! light level over the photoacclimation layer: (tmp_irrad_aclm/tmp_zaclm)*24/daylength
        cobalt%irr_aclm_inst(i,j,1:kblt(i,j)) = tmp_irrad_aclm/max(1.0e-6,tmp_zaclm)*24.0/ &
                                                max(cobalt%daylength(i,j),cobalt%min_daylength)
+       Issue: what would it take to remove irr_mix?
```
Comment (yichengt900) [inline generic_tracers/generic_COBALT.F90]: Missing `!`

**Comment by yichengt900** [comment]
Ok my patch fixes the model build process, but apparently the change  of [ml_aclm_efold](https://github.com/NOAA-CEFI-Regional-Ocean-Modeling/ocean_BGC/blob/143422f3f4c4eaa327c53f6bdfc281575ce93829/generic_tracers/generic_COBALT.F90#L6304)  and [eqs](https://github.com/NOAA-CEFI-Regional-Ocean-Modeling/ocean_BGC/blob/143422f3f4c4eaa327c53f6bdfc281575ce93829/generic_tracers/generic_COBALT.F90#L8374) changed our baseline answers and CI failed again. I will update our baselines.

<s> I will add a runtime parameter change to our CI test to maintain the original answers for now.<s> Once we confirm that this parameter change gives us satisfactory results from long-term hindcast run, I will update our baseline. @charliestock , are you okay with this approach?<s>

**Comment by yichengt900** [comment]
Oops, @charliestock plz ignore my previous comment. I missed your original comment. The changes are not only about the parameter itself, also it should be  the number of irradiance e-foldings that defined a deep mixed layer for photoacclimation. In this case we will have to update baselines anyway. I will do that.

**Review by yichengt900 — APPROVED** [review]
@charliestock, thanks for the code clean-up and improved commenting for the light and phytoplankton growth sections. This PR looks good to me now. We will have to consider conducting hindcast runs to test the impact of the new default value of `ml_aclm_efold` as well as the modified formulation

**Pull request merged**